lsp-iedit.el

```diff
@@ -73,5 +75,19 @@ See also `lsp-enable-symbol-highlighting'."
                                  (lsp--text-document-position-params))))
     (lsp-iedit--on-ranges (mapcar #'lsp:document-highlight-range highlights))))
 
+;;;###autoload
+(defun lsp-evil-multiedit-highlights ()
+  "Start an `evil-multiedit' operation on the documentHighlights at point.
+This can be used as a primitive `lsp-rename' replacement if the
+language server doesn't support renaming.
+
+See also `lsp-enable-symbol-highlighting'."
+  (interactive)
+  (require 'evil-multiedit)
+  (if (fboundp 'ahs-clear) (ahs-clear))
+  (setq evil-multiedit--dont-recall t)
```

Comment:
What does setting this variable achieve, and can it be done without depending on the internals of `evil-multiedit`? Perhaps something needs to be addressed upstream?

Reply:
Aside from that, LGTM. I have made an `iedit` PR to improve `iedit-lib-start`, so most of the set-up will be gone soon, hopefully.

Reply:
It is connected to `evil-multiedit-restore`. If this variable is `t`, then the selected symbols are not saved for recalling. I didn't choose to use this setting by my own, I followed how `evil-multiedit-match-all` sets things up. We need to use `t`, because restore doesn't work correctly if `lsp-evil-multiedit-highlights` did the highlighting. The reason for this is that it doesn't actually saves the selected symbols, but just the area and the regexp. So this feature doesn't work correctly even in non-lsp situations: for example, if the user selected symbols one-by-one, skipping some, then the skipped ones will be incorrectly re-selected at restore.

Reply:
I see. Perhaps a new command, `evil-multiedit-from-custom-iedit` or something like that should be added upstream. However, this needn't be addressed now.

Reply:
Do you need to set `evil-multiedit--dont-recall` or just let-bind is enough?

Reply:
It is need to be set because this variable is checked at a later time, in `evil-multiedit-abort` (when the mode is aborted, it is checked whether the current selections need to be saved so they can be recalled later).